framework/PluginsBaseClasses/PostProcessorPluginBase.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on March 10, 2021
+
+@author: wangc
+"""
+
+#External Modules---------------------------------------------------------------
+import abc
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PluginBase import PluginBase
+from Models.PostProcessors.PostProcessor import PostProcessor
+#Internal Modules End-----------------------------------------------------------
+
+class PostProcessorPluginBase(PostProcessor, PluginBase):
+  """
+    This class represents a specialized class from which each PostProcessor plugins must inherit from
+  """
+  # List containing the methods that need to be checked in order to assess the
+  # validity of a certain plugin. This list needs to be populated by the derived class
+  _methodsToCheck = ['getInputSpecification', 'handlePluginInput', 'initializePlugin', 'runPlugin']
+  entityType = 'PostProcessor'
+
+  ##################################################
+  # Methods to link internal PostProcessor methods
+  ##################################################
+  def __init__(self, runInfoDict):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    PluginBase.__init__(self)
+    PostProcessor.__init__(self, runInfoDict)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    PostProcessor._handleInput(self, paramInput)
+    self.handlePluginInput(paramInput)
+
+  def initialize(self, runInfo, inputs, initDict=None):
+    """
+      this needs to be over written if a re initialization of the model is need it gets called at every beginning of a step
```

Comment:
Can you add a sentence at the beginning to describe the function of this method?